pkg/log/logger.go

```diff
@@ -94,19 +95,37 @@ func newZapLoggerTo(destWriter io.Writer, level LogLevel, opts ...zap.Option) *z
 const tenantLoggerKey = "tenantID"
 
 // AddFieldsFromCtx will check if provided context contain tracing span and
-// if the span is currently recording. If so, it will add trace_id and span_id
-// to logged values. It will also add the tenant id to the logged values.
-func AddFieldsFromCtx(logger logr.Logger, ctx context.Context) logr.Logger {
+// if the span is currently recording. If so, it will call spanLogValuesProcessor
+// function if it's also present in the context. If not it will add trace_id
+// and span_id to logged values. It will also add the tenant id to the logged
+// values.
+func AddFieldsFromCtx(
+	logger logr.Logger,
+	ctx context.Context,
+	extensions context.Context,
```

Comment:
this is where just adding logger specific values to the `RuntimeContext` would be better I think

Reply:
If I'll have enough time today, I'll try to refactor it